inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/MongoDBSource.java

```diff
@@ -17,56 +17,154 @@
+                    profile.get(TaskConstants.TASK_MONGO_PASSWORD));
+            props.setProperty(String.valueOf(MongoDbConnectorConfig.DATABASE_INCLUDE_LIST), database);
+            props.setProperty(String.valueOf(MongoDbConnectorConfig.COLLECTION_INCLUDE_LIST), collection);
+            props.setProperty(String.valueOf(MongoDbConnectorConfig.SNAPSHOT_MODE), snapshotMode);
+
+            executor = Executors.newSingleThreadExecutor();
+            executor.execute(startDebeziumEngine());
+
+        } catch (Exception ex) {
+            stopRunning();
+            throw new FileException("error init stream for " + collection, ex);
+        }
+    }
+
+    private Runnable startDebeziumEngine() {
+        return () -> {
+            AgentThreadFactory.nameThread(getThreadName() + "debezium");
+            try (DebeziumEngine<ChangeEvent<String, String>> debeziumEngine = DebeziumEngine.create(Json.class)
+                    .using(props)
+                    .using(OffsetCommitPolicy.always())
+                    .notifying(this::handleConsumerEvent)
+                    .build()) {
+
+                debeziumEngine.run();
+            } catch (Throwable e) {
+                LOGGER.error("do run error in mongoDB debezium: ", e);
+            }
+        };
+    }
+
+    private void handleConsumerEvent(List<ChangeEvent<String, String>> records,
+            RecordCommitter<ChangeEvent<String, String>> committer) throws InterruptedException {
+        boolean offerSuc = false;
+        for (ChangeEvent<String, String> record : records) {
+            SourceData sourceData = new SourceData(record.value().getBytes(StandardCharsets.UTF_8), 0L);
+            while (isRunnable() && !offerSuc) {
+                offerSuc = debeziumQueue.offer(sourceData, 1, TimeUnit.SECONDS);
+            }
+            committer.markProcessed(record);
+        }
+        committer.markBatchFinished();
+    }
 
     @Override
     public List<Reader> split(TaskProfile conf) {
-        MongoDBReader mongoDBReader = new MongoDBReader();
-        List<Reader> readerList = Collections.singletonList(mongoDBReader);
-        sourceMetric.sourceSuccessCount.incrementAndGet();
-        return readerList;
+        return null;
     }
 
     @Override
     protected String getThreadName() {
-        return null;
+        return "mongo-source-" + taskId + "-" + instanceId;
     }
 
     @Override
     protected void printCurrentState() {
-
+        LOGGER.info("mongo collection is {}", collection);
     }
 
     @Override
     protected boolean doPrepareToRead() {
-        return false;
+        return true;
     }
 
     @Override
     protected List<SourceData> readFromSource() {
-        return null;
-    }
-
-    @Override
-    public Message read() {
-        return null;
+        List<SourceData> dataList = new ArrayList<>();
+        try {
+            SourceData sourceData = debeziumQueue.poll(1, TimeUnit.SECONDS);
```

Comment:
You can read multiple entries at once here, just limit the total length

Reply:
Done